hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetTypeManagement.java

```diff
@@ -122,54 +127,53 @@ public void delete(final Long targetTypeId) {
+    public Page<TargetType> findByRsql(final Pageable pageable, final String rsqlParam) {
         final Specification<JpaTargetType> spec = RSQLUtility.buildRsqlSpecification(rsqlParam, TargetTypeFields.class,
                 virtualPropertyReplacer, database);
 
         return convertPage(targetTypeRepository.findAll(spec, pageable), pageable);
     }
 
     @Override
-    public Optional<TargetType> get(long id) {
+    public Optional<TargetType> get(final long id) {
         return targetTypeRepository.findById(id).map(targetType -> targetType);
     }
 
     @Override
-    public Optional<TargetType> findByTargetId(long targetId) {
-        return targetTypeRepository
-                .findOne(TargetTypeSpecification.hasTarget(targetId)).map(TargetType.class::cast);
+    public Optional<TargetType> findByTargetId(final long targetId) {
+        return targetTypeRepository.findOne(TargetTypeSpecification.hasTarget(targetId)).map(TargetType.class::cast);
     }
 
     @Override
-    public List<TargetType> findByTargetIds(Collection<Long> targetIds) {
-        return targetTypeRepository.findAll(TargetTypeSpecification.hasTarget(targetIds))
-                .stream().map(TargetType.class::cast).collect(Collectors.toList());
+    public List<TargetType> findByTargetIds(final Collection<Long> targetIds) {
+        return targetTypeRepository.findAll(TargetTypeSpecification.hasTarget(targetIds)).stream()
+                .map(TargetType.class::cast).collect(Collectors.toList());
     }
 
     @Override
-    public Optional<TargetType> findByTargetControllerId(String controllerId) {
-        return targetTypeRepository
-                .findOne(TargetTypeSpecification.hasTargetControllerId(controllerId)).map(TargetType.class::cast);
+    public Optional<TargetType> findByTargetControllerId(final String controllerId) {
+        return targetTypeRepository.findOne(TargetTypeSpecification.hasTargetControllerId(controllerId))
+                .map(TargetType.class::cast);
     }
 
     @Override
-    public List<TargetType> findByTargetControllerIds(Collection<String> controllerIds) {
-        return targetTypeRepository.findAll(TargetTypeSpecification.hasTargetControllerIdIn(controllerIds))
-                .stream().map(TargetType.class::cast).collect(Collectors.toList());
+    public List<TargetType> findByTargetControllerIds(final Collection<String> controllerIds) {
+        return targetTypeRepository.findAll(TargetTypeSpecification.hasTargetControllerIdIn(controllerIds)).stream()
+                .map(TargetType.class::cast).collect(Collectors.toList());
     }
 
     @Override
-    public List<TargetType> get(Collection<Long> ids) {
+    public List<TargetType> get(final Collection<Long> ids) {
         return Collections.unmodifiableList(targetTypeRepository.findAllById(ids));
     }
 
     @Override
-    public TargetType update(TargetTypeUpdate update) {
+    public TargetType update(final TargetTypeUpdate update) {
```

Comment:
Transactional and Retryable annotations missing for update method

Reply:
Good catch! Added